parse_grid.py

```
@@ -165,7 +165,12 @@ def aggregate_appliance_availability(
 
 def _get_table_and_header(grid_html: str) -> tuple[Optional[Tag], Optional[Tag]]:
     """Extract main table and header row."""
-    soup = BeautifulSoup(grid_html, "lxml")
+    try:
+        soup = BeautifulSoup(grid_html, "lxml")
+    except Exception:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)